src/game_api/script/usertypes/game_manager_lua.cpp

```diff
@@ -95,20 +97,67 @@ void register_usertypes(sol::state& lua)
         &JournalPopupUI::timer,
         "slide_position",
         &JournalPopupUI::slide_position);
+    lua.new_usertype<InputDevice>(
+        "InputDevice",
+        "input_index",
+        &InputDevice::input_index,
+        "buttons",
+        &InputDevice::buttons);
     lua.new_usertype<GameProps>(
         "GameProps",
         "buttons",
-        &GameProps::buttons,
-        "buttons_extra",
-        &GameProps::buttons_extra,
+        [](GameProps& gp) -> uint32_t
+        {
+            return gp.buttons[1];
+        },
+        "buttons_previous",
+        [](GameProps& gp) -> uint32_t
+        {
+            return gp.buttons_previous[1];
+        },
+        "buttons_menu",
+        &GameProps::buttons_menu,
         "buttons_menu_previous",
         &GameProps::buttons_menu_previous,
-        "buttons_menu",
+        "input",
+        &GameProps::buttons,
+        "input_previous",
+        &GameProps::buttons_previous,
+        "input_menu",
         &GameProps::buttons_menu,
+        "input_menu_previous",
+        &GameProps::buttons_menu_previous,
```

Comment:
why `buttons_menu` and `input_menu` ?

Reply:
I may have copypasted too many things. Yeah I just wanted to deprecate stupid singular buttons and couldn't use the same name and didn't want to call the array buttonses like gollum.